packages/features/insights/server/trpc-router.ts

```diff
@@ -914,7 +914,7 @@ export const insightsRouter = router({
     };
 
     // Validate if user belongs to org as admin/owner
-    if (user.organizationId) {
+    if (user.organizationId && user.organization.isOrgAdmin) {
```

Comment:
This fixes few issues, such as the unauthorized error and displaying all teams inside an org for an org member